feature/security/gnsi/authz/tests/authz_test.go

```diff
@@ -0,0 +1,752 @@
+	}
+	if !cmp.Equal(&newpolicy, finalPolicy) {
+		t.Fatalf("Not Expecting Policy Mismatch before and after the Wait):\n%s", cmp.Diff(&newpolicy, finalPolicy))
+	}
+}
+
+// Authz-4 Reboot Persistent
+func TestAuthz4(t *testing.T) {
+	// Pre-Test Section
+	dut := ondatra.DUT(t, "dut")
+	_, policyBefore := authz.Get(t, dut)
+	t.Logf("Authz Policy of the Device %s before the Reboot Trigger is %s", dut.Name(), policyBefore.PrettyPrint())
+	defer policyBefore.Rotate(t, dut, uint64(time.Now().UnixMilli()), fmt.Sprintf("v0.%v", (time.Now().UnixNano())), false)
+
+	// Fetch the Desired Authorization Policy and Attach base Admin Policy Before Rotate
+	policies := loadPolicyFromJsonFile(t, dut, "testdata/policy.json")
+	newpolicy := getPolicyByName(t, "policy-normal-1", policies)
+	newpolicy.AddAllowRules("base", []string{*testInfraID}, []*gnxi.RPC{gnxi.RPCs.ALL})
+	expCreatedOn := uint64(time.Now().UnixMilli())
+	expVersion := fmt.Sprintf("v0.%v", (time.Now().UnixNano()))
+	t.Logf("New Authz Policy is %s", newpolicy.PrettyPrint())
+	newpolicy.Rotate(t, dut, expCreatedOn, expVersion, false)
+
+	// Trigger Section - Reboot
+	gnoiClient, err := dut.RawAPIs().BindingDUT().DialGNOI(context.Background())
+	if err != nil {
+		t.Fatalf("Failed to connect to gnoi server, err: %v", err)
+	}
+	rebootRequest := &gnps.RebootRequest{
+		Method: gnps.RebootMethod_COLD,
+		Force:  true,
+	}
+	bootTimeBeforeReboot := gnmi.Get(t, dut, gnmi.OC().System().BootTime().State())
+	t.Logf("DUT boot time before reboot: %v", bootTimeBeforeReboot)
+	var currentTime string
+	currentTime = gnmi.Get(t, dut, gnmi.OC().System().CurrentDatetime().State())
+	t.Logf("Time Before Reboot : %v", currentTime)
+	rebootResponse, err := gnoiClient.System().Reboot(context.Background(), rebootRequest)
+	t.Logf("Got Reboot response: %v, err: %v", rebootResponse, err)
+	if err != nil {
+		t.Fatalf("Failed to reboot chassis with unexpected err: %v", err)
+	}
+	for {
+		if errMsg := testt.CaptureFatal(t, func(t testing.TB) {
+			currentTime = gnmi.Get(t, dut, gnmi.OC().System().CurrentDatetime().State())
+		}); errMsg != nil {
+			t.Log("Reboot is started")
+			break
+		}
+		t.Log("Wait for reboot to be started")
+		time.Sleep(30 * time.Second)
+	}
+	startReboot := time.Now()
+	t.Logf("Wait for DUT to boot up by polling the telemetry output.")
+	for {
+		t.Logf("Time elapsed %.2f seconds since reboot started.", time.Since(startReboot).Seconds())
+		if errMsg := testt.CaptureFatal(t, func(t testing.TB) {
+			currentTime = gnmi.Get(t, dut, gnmi.OC().System().CurrentDatetime().State())
+		}); errMsg != nil {
+			t.Logf("Got testt.CaptureFatal errMsg: %s, keep polling ...", *errMsg)
+		} else {
+			t.Logf("Device rebooted successfully with received time: %v", currentTime)
+			break
+		}
+		if uint64(time.Since(startReboot).Seconds()) > maxRebootTime {
+			t.Fatalf("Check boot time: got %v, want < %v", time.Since(startReboot), maxRebootTime)
+		}
+	}
+	// Verification Section
```

Comment:
I think this section missed the policy validation

```
### Authz-4 reboot persistent

  1. Use `gNSI.Rotate` method to push and finalize policy `policy-normal-1`.
  2. Reboot the device.
  3. Reconnect to the device, issue `gNSI.Get` and `gNMI.Get` and validate the value of `version`, `created_on` and gRPC policy content does not change.
  4. Ensure actual corresponding clients are authorized per the the above table for policy `policy-normal-1`.
```

Reply:
Done